submariner/IPV6-datapath.md

```diff
@@ -0,0 +1,185 @@
+# Submariner Enhancement for IPV6 datapath
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+IPv4, IPV6 and dual-stack networking is supported for Kubernetes cluster starting in 1.21.
+IPV6 networking allowing assignment of IPv6 addresses.
+Dual-stack networking allowing the simultaneous assignment of both IPv4 and IPv6 addresses.
+
+IPv4/IPv6 dual-stack on your Kubernetes cluster provides the following features:
+
+* Dual-stack Pod networking (a single IPv4 and IPv6 address assignment per Pod)  
+* IPv4 and IPv6 enabled Services  
+* Pod off-cluster egress routing (eg. the Internet) via both IPv4 and IPv6 interfaces
+
+Currently, Submariner only supports IPV4 datapath and this proposal explains the changes required to support IPV6 and dual-stack clusters.
+
+## Overview
+
+Submariner’s control plane uses a central broker component to facilitate the exchange of metadata information between participating clusters.
+
+Both inter-cluster datapath and service-discovery (Lighthouse) apply their configuration based on the information received from control plane.
+
+The Broker, inter-cluster datapath and Lighthouse should be updated to support mixed clusters environment (e.g: dual-stack, V6 only , V4 only).
+
+### Control plane
+
+For mixed clusters environment we may experience connectivity issue.
+Some clusters and broker may not connect if the Broker’s cluster doesn’t support dual-stack.
```

Comment:
This statement implies that Broker cluster **has** to support dualStack in mixed env. I think this is not required - f.e., if all the managed clusters support dualStack then having a Broker cluster that just supports IPv4 alone should be sufficient isnt it?
The Gateways on the managed clusters can connect to Broker over IPv4 and still exchange metadata that includes info of V4 and V6.

Reply:
Correct, will rephrase